### PR TITLE
Add file upload route

### LINKED ISSUE #5: Allow for saving of file form uploads to disk
Add a route handler for accepting the uploaded file and write it to disk

### PR DESCRIPTION
**Pull request by cailyncodes**
Should close #5

### CODE REVIEW

File: src/data/users.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { User } from "../models/user";
+
+export const users: User[] = [
+	{
+		id: 1,
+		wingedKeysId: "2c0ec653-8829-4aa1-82ba-37c8832bbb88",
+		firstName: "Voldy",
```
Comment (mmazanec22) [inline src/data/users.ts:7]: Lol, yes

File: src/middleware/authenticate.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	requestProperty: 'token'
+});
+
+const handleError = (
+	err: Error,
+	_: Request,
+	res: Response,
+	next: NextFunction
+) => {
+	if (err.name === 'UnauthorizedError') {
```
Comment (jlhogan) [inline src/middleware/authenticate.ts:20]: nit: weird whitespace

File: src/types/custom.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+declare namespace Express {
```
Comment (jlhogan) [inline src/types/custom.d.ts:1]: why is this a custom type?

Comment (cailyncodes) [reply src/types/custom.d.ts:1]: B/c of Typescript, we need to define that we are adding `token` and `sub` to the request properties

**Pull request merged**